seqr/views/apis/individual_api.py

```diff
@@ -785,22 +786,41 @@ def save_individuals_metadata_table_handler(request, project_guid, upload_file_i
 
     return create_json_response(response)
 
+
 @login_and_policies_required
 def get_individual_rna_seq_data(request, individual_guid):
     individual = Individual.objects.get(guid=individual_guid)
     check_project_permissions(individual.family.project, request.user)
-    outlier_data = RnaSeqOutlier.objects.filter(sample__individual=individual, sample__is_active=True)
 
-    rna_seq_data = {
-        data['geneId']: data for data in get_json_for_rna_seq_outliers(outlier_data)
+    outlier_data = {
+        'outliers': _get_rna_seq_data(RnaSeqOutlier, individual),
+        'spliceOutliers': _get_rna_seq_data(RnaSeqSpliceOutlier, individual,
+                                            max_significant_num_per_tissue=MAX_SIGNIFICANT_OUTLIER_NUM),
     }
-    genes_to_show = get_genes([gene_id for gene_id, data in rna_seq_data.items() if data['isSignificant']])
+
+    genes_to_show = get_genes({
+        gene_id for rna_data in outlier_data.values() for gene_id, data in rna_data.items() if any([d['isSignificant'] for d in data])
+    })
 
     return create_json_response({
-        'rnaSeqData': {individual_guid: {'outliers': rna_seq_data}},
+        'rnaSeqData': {individual_guid: outlier_data},
         'genesById': genes_to_show,
     })
 
+
+def _get_rna_seq_data(model_cls, individual, **kwargs):
+    outlier_data = model_cls.objects.filter(sample__individual=individual, sample__is_active=True)
+
+    outliers_by_gene = defaultdict(list)
+    for data in get_json_for_rna_seq_outliers(
+            outlier_data,
+            nested_fields=[{'fields': ('sample', 'tissue_type'), 'key': 'tissueType'}],
+            **kwargs):
```

Comment:
Per discussion in the other PR, the kwargs and `nested_fields` arguments should be moved directly into `get_json_for_rna_seq_outliers` so the function can be better shared. If the behavior does need to differ from other places this function is called, add a flag to it to make it explicit

Reply:
The `get_json_for_rna_seq_outliers` function is also used in some test cases. I can eliminate them using the `get_sjon_for_queryset` or the `value_list`. But I would like to leave the change to the `RnaSeq splice` tag ticket because it updates the `rnaSeqData` in the variant search results. And the search results need to be updated for the other ticket.

Reply:
Using helper functions in tests in a way they are never used in the code is bad practice. You should make helper functions that are actually designed for real use cases, and then adjust test accordingly. `get_json_for_rna_seq_outliers` needs to have consistancy in how its called in the code after this PR is merged, it can;t be overly configurable and inconsistent here and then cleaned up later